src/visitable.cpp

```diff
@@ -72,14 +72,24 @@ bool visitable<T>::has_item_with( const std::function<bool( const item & )> &fil
     } ) == VisitResponse::ABORT;
 }
 
+// Sums the two terms, being careful to not trigger any overflows.
+// Doesn't handle underflow.
+static int sum_no_wrap( int a, int b ) {
```

Comment:
Doygen style comments could be useful if you undertake the above change (which isn't essential for this PR)